StatsdClient/Statsd.cs

```diff
@@ -204,13 +234,33 @@ private void InitialiseInternal(Func<IOutputChannel> createOutputChannel, string
         /// <param name="prefix"></param>
         /// <param name="value"></param>
         /// <param name="postFix">A value to append to the end of the line.</param>
-        /// <returns>The formatted metric</returns>
-        protected virtual string PrepareMetric(string metricType, string name, string prefix, string value, string postFix = null)
-        {
-            return (String.IsNullOrEmpty(prefix) ? name : (prefix + "." + name))
-                   + ":" + value
-                   + "|" + metricType
-                   + (postFix == null ? String.Empty : "|" + postFix);
+        /// <returns>The formatted metric</returns>
+        protected virtual string PrepareMetric(string metricType, 
+            string name, string prefix, string value, string namePostfix = null, string postFix = null)
+        {
+            StringBuilder builder = new StringBuilder();
```

Comment:
I'm just worried that a string builder will be slower. Can you write a test that does 100k iterations over this and see the difference. When I was working on this I found string concat to be much faster and you don't need to do the null check.